### PR TITLE
[core] Refacto sorting comparator

### PR DESCRIPTION
**Pull request by flaviendelangle**
The goal here is to improve the comprehension of the codebase for future developers.

- [x] Like for filtering, expose a single very simple method to be called in the sorting pre-processors
- [x] Add sections `API METHODS`, `EVENTS`, `PRE-PROCESSINGS`, .... on `useGridSorting` 
- [x] Move the `buildAggregatedSortingApplier` into a standalone file to lighten the main hook

@alexfauquette I'm interested if anything is not crystal clear here.

### CODE REVIEW

File: packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+      return { getSortCellParams, comparator };
+    })
+    .filter((comparator): comparator is GridSortingFieldComparator => !!comparator);
+
+  if (comparatorList.length === 0) {
+    return null;
+  }
+
+  const aggregatedComparator = (row1: GridSortCellParams[], row2: GridSortCellParams[]) => {
```
Comment (flaviendelangle) [inline packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts:53]: I did not change this code, but I think we could make this `reduce` easier to read.

File: packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ * Generates a method to easily sort a list of rows according to the current sort model.
+ * @param {GridSortModel} sortModel The model with which we want to sort the rows.
+ * @param {GridApiRef} apiRef The API of the grid.
+ * @returns {GridSortingModelApplier | null} A method that sorts a list of rows according to the current sort model. If `null`, we consider that the rows should remain in the order there were provided.
+ */
+export const buildAggregatedSortingApplier = (
+  sortModel: GridSortModel,
+  apiRef: GridApiRef,
+): GridSortingModelApplier | null => {
+  const comparatorList = sortModel
```
Comment (alexfauquette) [inline packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts:27]:
```suggestion
  // Compute the list of comparator sorted by priority
  const comparatorList = sortModel
```

File: packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      });
+
+      return { getSortCellParams, comparator };
+    })
+    .filter((comparator): comparator is GridSortingFieldComparator => !!comparator);
+
+  if (comparatorList.length === 0) {
+    return null;
+  }
+
```
Comment (alexfauquette) [inline packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts:52]:
```suggestion
  // Compare two rows according to comparatorList
```

File: packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  const aggregatedComparator = (row1: GridSortCellParams[], row2: GridSortCellParams[]) => {
+    return comparatorList.reduce((res, colComparator, index) => {
+      if (res !== 0) {
+        return res;
```
Comment (alexfauquette) [inline packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts:56]:
```suggestion
        // return the results of the first comparator which distinguish the two rows
        return res;
```

File: packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+type GridSortingFieldComparator = {
+  getSortCellParams: (id: GridRowId) => GridSortCellParams;
+  comparator: GridComparatorFn;
+};
+
+/**
+ * Generates a method to easily sort a list of rows according to the current sort model.
+ * @param {GridSortModel} sortModel The model with which we want to sort the rows.
+ * @param {GridApiRef} apiRef The API of the grid.
+ * @returns {GridSortingModelApplier | null} A method that sorts a list of rows according to the current sort model. If `null`, we consider that the rows should remain in the order there were provided.
```
Comment (alexfauquette) [inline packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts:21]: I've been confused by the returned function. It does not seem to sort the list. It provides the list of row ids sorted according to the sort model

**Review by alexfauquette — COMMENTED** [review]
It seems clearer thanks to the complex map and reduce operation put into a specific file.

This `buildAggregatedSortingApplier` is a bit hard to read because it contains two complex array manipulation which are them self preliminary computation for the result.

I proposed to add some comments to simplify the reading. such that reader know "this is a step, this is another one, this is where step are merge".

Another solution would be to define manipulation outside of the main function and so 

```js
const comparatorList = sortModel
				.map((item) => getComparator(apiRef)(item))
				.filter(isGridSortingFieldComparator)
```

**Comment by flaviendelangle** [comment]
@alexfauquette I moved the two main functions of `buildAggregatedSortingApplier` into standalone functions with full JSDoc :+1:

**Review by alexfauquette — APPROVED** [review]
Sorry, I forget to put the review last time

**Pull request merged**